### ISSUE TITLE
Does `SXTMap` work?

### ISSUE THREAD

**Issue by ayshih** [sunpy/sunpy#3440]
I recently stumbled across the fact that we apparently don't have a test file for `SXTMap` (Yohkoh/SXT data).  It looks one would have been added in the now-closed PR #2056.  However, that PR thread seems to indicate that our `SXTMap` class cannot read the SXT files from VSO because of some format issue with the `dateobs` header keyword.  Does `SXTMap` currently work?

See also #664, but that issue is more about adding tests than whether the functionality works in the first place.

**Comment by wtbarnes** [comment]
I looked into this a bit. First, I attempted to query the VSO for some SXT data and downloaded the first full-disk image

```python
from sunpy.net import Fido,attrs
q = Fido.search(attrs.Instrument('sxt'), attrs.Time('1991-11-14','1991-11-15'))
f = Fido.fetch(q[0,0])
```

When I tried to construct a Map from this file, I initially got an `UnrecognizedFileTypeError`, presumably because these files do not have the '.fits' extension. I added '.fits' to the end of the filename to force `Map` to use the FITS reader and got an `OSError: Header missing END card`. Whatever these files are, they don't seem to want to play nice with the astropy FITS reader. Maybe I've messed up the VSO query and am getting the wrong files back?

Next, I downloaded this file (https://umbra.nascom.nasa.gov/yohkoh/data/sxt_fits/level_1/full_sun/1991/11/05/sxtf_19911105_111024_221.fts) directly from the SXT archive. This is a Level 1, full-sun observation. I can construct a Map from this file and plot it with `peek`.

However, the `wavelength` property throws an error because the 'WAVELNTH' entry in the header is a string (it's the name of the filter wheel I think?) and `Map` expects it to be something that can be expressed as a `Quantity`. Interestingly, in [`SXTMap.measurement`, we manipulate `meta['wavelnth']` assuming it is a string](https://github.com/sunpy/sunpy/blob/master/sunpy/map/sources/yohkoh.py#L70). I suspect that the `Quantity` requirement on the `wavelength` property was added later and we didn't update the logic in `measurement`.

**tl;dr I cannot query a usable SXT file from the VSO, but it is possible to construct a Map from a level 1 FITS file from the archive. However, these still have issues and our `SXTMap` still needs some work.**

**Comment by dstansby** [comment]
I think the remaining TODO here is add a SXTMap source test? That's tracked in https://github.com/sunpy/sunpy/issues/5140, so I'll close this as a file from the SXT archive seems to be working, but feel free to re-open if I've missed anything.
